src/Session.ts

```diff
@@ -192,6 +192,12 @@ export class Session implements SessionMetadata {
     return this.#storage.sessionNumAttempts(this._id);
   }
 
+  // function to get all attempts in a session. Can be very slow for large sessions
+  async getAllAttempts(): Promise<StoredAttempt[]> {
```

Comment:
I explicitly want to avoid any inefficient APIs. Is there something specific you're trying to get that doesn't necessarily require the client to return the full list of attempts? (For example, stats like "best avg of X" can be cached without iterating over the whole DB every time.)